cxx/non-atomic-action-interpreter-module/test/agent/CheckDynamicArgumentTestAgent.cpp

```diff
@@ -1,46 +1,28 @@
-#include "sc-agents-common/keynodes/coreKeynodes.hpp"
-#include "sc-agents-common/utils/AgentUtils.hpp"
+#include "CheckDynamicArgumentTestAgent.hpp"
 
 #include "test/keynodes/TestKeynodes.hpp"
 
-#include "CheckDynamicArgumentTestAgent.hpp"
-
 using namespace nonAtomicActionInterpreterModuleTest;
 
-SC_AGENT_IMPLEMENTATION(CheckDynamicArgumentTestAgent)
+ScResult CheckDynamicArgumentTestAgent::DoProgram(ScAction & action)
 {
-  if (!edgeAddr.IsValid())
-  {
-    return SC_RESULT_ERROR;
-  }
-
-  ScAddr actionAddr = m_memoryCtx.GetEdgeTarget(edgeAddr);
-
-  ScIterator3Ptr iterator3Ptr = m_memoryCtx.Iterator3(
-      TestKeynodes::check_dynamic_argument_test_action, ScType::EdgeAccessConstPosPerm, actionAddr);
-  if (!iterator3Ptr->Next())
-  {
-    return SC_RESULT_OK;
-  }
```

Comment:
Looks useful

Reply:
sc-machine performs this check because GetActionClass returns check_dynamic_argument_test_action